tests_e2e/playwright/navigator.py

```diff
@@ -0,0 +1,448 @@
+import logging
+import re
+import time
+
+from playwright.sync_api import expect, sync_playwright
+
+logger = logging.getLogger()
+
+
+class Navigator:
+    """Base class for Nebari Playwright testing. This provides setup and
+    teardown methods that all tests will need and some other generally useful
+    methods such as clearing the workspace. Specific tests such has "Run a
+    notebook" are included in separate classes which use an instance of
+    this class.
+
+    The Navigator class and the associated test classes are design to be able
+    to run either standalone, or inside of pytest. This makes it easy to
+    develop new tests, but also have them fully prepared to be
+    included as part of the test suite.
+
+    Parameters
+    ----------
+    nebari_url: str
+        Nebari URL to access for testing
+    username: str
+        Login username for Nebari. For Google login, this will be email address.
+    password: str
+        Login password for Nebari. For Google login, this will be the Google
+        password.
+    auth: str
+        Authentication type of this Nebari instance. Options are "google" and
+        "password".
+    headless: bool
+        (Optional) Run the tests in headless mode (without visuals). Defaults
+        to False.
+    slow_mo: int
+        (Optional) Additional milliseconds to add to each Playwright command,
+        creating the effect of running the tests in slow motion so they are
+        easier for humans to follow. Defaults to 0.
+    browser: str
+        (Optional) Browser on which to run tests. Options are "chromium",
+        "webkit", and "firefox". Defaults to "chromium".
+    instance_name: str
+        (Optional) Server instance type on which to run tests. Options are
+        based on the configuration of the Nebari instance. Defaults to
+        "small-instance". Note that special characters (such as parenthesis)
+        will need to be converted to dashes. Check the HTML element to get the
+        exact structure.
+    video_dir: None or str
+        (Optional) Directory in which to save videos. If None, no video will
+        be saved. Defaults to None.
+    """
+
+    def __init__(
+        self,
+        nebari_url,
+        username,
+        password,
+        auth,
+        headless=False,
+        slow_mo=0,
+        browser="chromium",
+        instance_name="small-instance",
+        video_dir=None,
+    ):
+        self.nebari_url = nebari_url
+        self.username = username
+        self.password = password
+        self.auth = auth
+        self.initialized = False
+        self.headless = headless
+        self.slow_mo = slow_mo
+        self.browser = browser
+        self.instance_name = instance_name
+        self.video_dir = video_dir
+
+        self.setup(
+            browser=self.browser,
+            headless=self.headless,
+            slow_mo=self.slow_mo,
+        )
+        self.wait_for_server_spinup = 30000  # 5 * 60 * 1000  # 5 minutes in ms
```

Comment:
This is missing a `0`. As is, we have a timeout for 30 seconds or 0.5 minutes.

```suggestion
        self.wait_for_server_spinup = 300_000  # 5 * 60 * 1_000  # 5 minutes in ms
```